.github/workflows/remote-bazel-build.yaml

```diff
@@ -16,13 +16,15 @@ jobs:
         with:
           token: ${{ secrets.BUILDBUDDY_GITHUB_USER_TOKEN }}
 
-      - name: Download bb cli
```

Comment:
Just a reminder to revert this before you merge

Reply:
side question: why don't we test the CLI from the current HEAD?

This currently make the "remote bazel" failures in each PR not very actionable 🤔

Reply:
Hmm that's a good point. It's slow to build the cli on each run, so maybe we can detect if cli-specific files have changed before building. I can add that separately